src/lib/fields/DatesField.js

```diff
@@ -0,0 +1,41 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2020 CERN.
+// Copyright (C) 2020 Northwestern University.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import _get from 'lodash/get';
+import _isEqual from 'lodash/isEqual';
+
+import { Field } from './Field';
+
+
+export class DatesField extends Field {
+
+  /**
+   * Serialize frontend dates to backend dates.
+   * Strips it out if only default entry is there.
+   * @method
+   * @param {object} record - in frontend format
+   * @returns {object} record - in API format
+   */
+  serialize(record) {
+    const recordDates = _get(
+      record,
+      this.fieldpath,
+      this.serializedDefault
+    );
+    // Remove dates if only the default empty date is there
+    // NOTE: We have to do this because a date is show for UI sake, but if not
+    //       filled, it would be sent to backend and generate a validation
+    //       error
+    // NOTE: This check is because null values are stripped at first.
+    //       We might want to revisit that...
+    if ( _isEqual(recordDates, [{type: 'accepted'}]) ) {
+      delete record.metadata.dates;
```

Comment:
I don't know how I feel about this yet... We know the record passed is already a clone + faster, but maybe this might change...